server/src/main/java/org/opensearch/index/compositeindex/datacube/Dimension.java

```diff
@@ -45,4 +46,24 @@ public interface Dimension extends ToXContent {
     List<String> getSubDimensionNames();
 
     DocValuesType getDocValuesType();
+
+    default ComparisonType getComparisonType() {
+        return ComparisonType.SIGNED;
+    }
+
+    default Comparator<Long> comparator() {
```

Comment:
Lets return a different comparator based on type 

Lets avoid this check 
```
getComparisonType() == ComparisonType.UNSIGNED ? Long.compareUnsigned(a, b) : Long.compare(a, b);
```

Let the individual comparators explicitly have specific comparisons only.

It'll scale and it'll be more performant.

Reply:
> Let the individual comparators explicitly have specific comparisons only.

Done. We now have an ENUM - `ComparatorType`, which can have 2 values `LONG` and `UNSIGNED_LONG` and they explicitly perform their respective comparisions.